Fixed #33878 -- Switched to system fonts

knyghty opened the pull request:
Changed in the admin, and the default_urlconf template. Could possibly add them to the technical error pages, but they don't currently specify fonts at all other than to swtich to monospace.

Copied the stacks from wagtail, they should counteract some of the drawbacks mentioned in the ticket.

knyghty commented on django/contrib/admin/static/admin/css/base.css:
```diff
@@ -2,60 +2,89 @@
 /* VARIABLE DEFINITIONS */
 html[data-theme="light"],
 :root {
-  --primary: #79aec8;
```
I fixed the indentation here as it's different from the rest of the file, but I can revert this or put in another commit maybe.

carltongibson replied:
Yes please. (See other comment 🙂)

knyghty replied:
I moved to a separate commit as below.

knyghty commented on django/contrib/admin/static/admin/css/base.css:
```diff
@@ -2,60 +2,89 @@
+        "Segoe UI Emoji",
+        "Segoe UI Symbol",
+        "Noto Color Emoji";
+    --font-family-monospace:
+        ui-monospace,
```
I wasn't really sure how to indent this, not sure we have anything else in CSS spanning multiple lines.

carltongibson commented on django/contrib/admin/static/admin/css/base.css:
```diff
@@ -2,60 +2,89 @@
-
-  --header-color: #ffc;
-  --header-branding-color: var(--accent);
-  --header-bg: var(--secondary);
-  --header-link-color: var(--primary-fg);
-
-  --breadcrumbs-fg: #c4dce8;
-  --breadcrumbs-link-fg: var(--body-bg);
-  --breadcrumbs-bg: var(--primary);
-
-  --link-fg: #447e9b;
-  --link-hover-color: #036;
-  --link-selected-fg: #5b80b2;
-
-  --hairline-color: #e8e8e8;
-  --border-color: #ccc;
-
-  --error-fg: #ba2121;
-
-  --message-success-bg: #dfd;
-  --message-warning-bg: #ffc;
-  --message-error-bg: #ffefef;
-
-  --darkened-bg: #f8f8f8; /* A bit darker than --body-bg */
-  --selected-bg: #e4e4e4; /* E.g. selected table cells */
-  --selected-row: #ffc;
-
-  --button-fg: #fff;
-  --button-bg: var(--primary);
-  --button-hover-bg: #609ab6;
-  --default-button-bg: var(--secondary);
-  --default-button-hover-bg: #205067;
-  --close-button-bg: #888; /* Previously #bbb, contrast 1.92 */
-  --close-button-hover-bg: #747474;
-  --delete-button-bg: #ba2121;
-  --delete-button-hover-bg: #a41515;
-
-  --object-tools-fg: var(--button-fg);
-  --object-tools-bg: var(--close-button-bg);
-  --object-tools-hover-bg: var(--close-button-hover-bg);
```
Can we avoid the indentation change to keep the diff clean? (If there's a case for it, we can address separately).

carltongibson commented on django/contrib/admin/static/admin/css/base.css:
```diff
@@ -2,60 +2,89 @@
+    --object-tools-bg: var(--close-button-bg);
+    --object-tools-hover-bg: var(--close-button-hover-bg);
+
+    --font-family-primary:
```
These new variables may be worth calling out. (Do we mention the CSS vars at all in the customising the Admin theme docs 🤔)

(If I wanted to revert I'd override these with the old values right...)

knyghty replied:
All I can think of is this section: https://docs.djangoproject.com/en/4.0/ref/contrib/admin/#theming-support - I could change "The admin uses CSS variables to define colors." to "The admin uses CSS variables to define colors and fonts" or something like that.

I'm not personally convinced by having a list of variables in the docs, I think it will be a little painful to make sure they're in sync. I think anyone wanting to change the admin CSS is likely to be comfortable looking at the base CSS files to see the variable names and defaults. I'm not dead set on this though, I can see merits of documenting them, such as the version of docs being correct for your version - you don't have to hunt through github or your local file system to find the right file.

I am not really sure 🤔

carltongibson replied:
I think that small adjustment is likely fine. 

Certainly not listing the vars no. 

We already point to the file (and it may be linked to GitHub on `main` now) so I think that's enough. 

(But the release note calling it out too please. 🙂)

Thanks.

carltongibson reviewed (commented):
Hey @knyghty — thanks for this, let me have a look at it. 👍 

Just quickly though, I think this would merit an entry in the release notes (`4.2.txt`)

carltongibson commented:
Here's the relevant docs section: https://docs.djangoproject.com/en/4.1/ref/contrib/admin/#theming-support

carltongibson reviewed (commented):
Hey @knyghty — This looks good as far as I can see. 

Thanks for the various links to the previous discussions elsewhere! That really helps. 
I think following Wagtail is good here: if problems do arise there's an opportunity to share learnings. 

If we add some notes to the docs about the font family variables — likely just mentioning them is enough — then folks can always override if they need to. 

Please do move the indentation change to a separate commit (or adjust to keep it, as you think, but yes, we use 4 spaces elsewhere...) 

@thibaudcolas Any comments?

knyghty commented:
I moved the indentation change to a separate commit, tweaked docs and added a release note 👍🏻

carltongibson commented:
buildbot, test on selenium.

carltongibson reviewed (approved):
Great @knyghty — Let's go with it! 🎁

I've just squashed in the Docs commit with a minor edit. 

Thanks again for taking this on.

knyghty commented:
buildbot, test on selenium.

knyghty commented:
@carltongibson okay, I fixed that (and a small issue where I missed a line break between the first two fonts) and squashed, hopefully ready to go now.
